src/pages/Wallet/SendPage.tsx

```
@@ -97,32 +101,83 @@ const SendPage = () => {
+      setIsSending(false)
+    }
+  }
 
-        addPendingTx({
-          txId: txSendResp.data.txId,
-          toAddress: address,
-          timestamp: new Date().getTime(),
-          amount: fullAmount
-        })
+  const consolidateUTXOs = async () => {
+    if (client && wallet) {
+      setIsConsolidating(true)
+      try {
+        const txCreateResp = await client.clique.transactionConsolidateUTXOs(
+          wallet.publicKey,
+          wallet.address,
+          wallet.address
+        )
+        for (const { txId, unsignedTx } of txCreateResp.data.unsignedTxs) {
+          const txSendResp = await signAndSendTransaction(txId, unsignedTx, wallet.address, wallet.privateKey)
+          if (txSendResp) {
+            addPendingTx({
+              txId: txSendResp.data.txId,
+              toAddress: wallet.address,
+              timestamp: new Date().getTime(),
+              amount: '',
+              type: 'consolidation'
+            })
+          }
+        }
 
-        setSnackbarMessage({ text: 'Transaction sent!', type: 'success' })
-        history.push('/wallet')
+        setSnackbarMessage({
+          text: 'Consolidation process in progress. You can try to send your transaction again once the pending consolidation transactions have been confirmed.',
+          type: 'info',
+          duration: 15000
+        })
+        setIsConsolidateUTXOsModalVisible(false)
+        initialOnModalClose.current()
       } catch (e) {
         setSnackbarMessage({
-          text: getHumanReadableError(e, 'Error while sending the transaction'),
+          text: getHumanReadableError(e, 'Error while consolidating UTXOs'),
           type: 'alert',
           duration: 5000
         })
       }
-
-      setIsSending(false)
+      setIsConsolidating(false)
     }
   }
 
+  const signAndSendTransaction = async (txId: string, unsignedTx: string, toAddress: string, privateKey: string) => {
+    if (!client) return
```

Should we throw an error if `client` is not defined? Or we could assume that `client` is define when this function is called

---

We can assume it's defined, because without a `client`, the user wouldn't be able to login to the wallet.